src/Simplify.elm

```diff
@@ -6370,6 +6383,89 @@ indexAccessChecks collection checkInfo n =
                 Nothing
 
 
+setChecks : CollectionProperties (FromListProperties (IndexableProperties {})) -> CheckInfo -> Maybe (Error {})
+setChecks collection checkInfo =
+    firstThatConstructsJust
+        [ \() ->
+            Maybe.andThen
+                (\emptiableArg -> callOnEmptyReturnsEmptyCheck emptiableArg collection checkInfo)
+                (thirdArg checkInfo)
+        , \() ->
+            case Evaluate.getInt checkInfo checkInfo.firstArg of
+                Just n ->
+                    case secondArg checkInfo of
+                        Just replacementArg ->
+                            if n < 0 then
+                                Just
+                                    (alwaysReturnsLastArgError
+                                        (qualifiedToString checkInfo.fn ++ " with negative index")
+                                        { lastArg = thirdArg checkInfo
+                                        , lastArgRepresents = collection.represents
+                                        }
+                                        checkInfo
+                                    )
+
+                            else
+                                setOnKnownElementChecks collection checkInfo n (Node.range replacementArg)
+
+                        Nothing ->
+                            Just
+                                (Rule.errorWithFix
+                                    { message = qualifiedToString checkInfo.fn ++ " with negative index will always return the same given " ++ collection.represents
+                                    , details =
+                                        [ "You can replace this call by \\_ -> identity." ]
+                                    }
+                                    checkInfo.fnRange
+                                    [ Fix.replaceRangeBy checkInfo.parentRange "\\_ -> identity" ]
```

Comment:
All good points :+1: 
For the `if`, it used to be in the correct location, and then I changed it. Seems like I'm missing a test case, so that's good to discover.
(And doh, didn't think of using `always`  *face-palm*)